lib/core/data/payment/payment_repository_impl.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:app/core/data/event/gql/event_tickets_query.dart';
+import 'package:app/core/data/payment/dtos/payment_card_dto/payment_card_dto.dart';
+import 'package:app/core/domain/payment/entities/payment_card_entity/payment_card_entity.dart';
+import 'package:app/core/domain/payment/payment_repository.dart';
+import 'package:app/core/failure.dart';
+import 'package:app/core/utils/gql/gql.dart';
+import 'package:app/injection/register_module.dart';
+import 'package:dartz/dartz.dart';
+import 'package:graphql_flutter/graphql_flutter.dart';
+import 'package:injectable/injectable.dart';
+
+@LazySingleton(as: PaymentRepository)
+class PaymentRepositoryImpl extends PaymentRepository {
+  final _client = getIt<AppGQL>().client;
+
+  @override
+  Future<Either<Failure, PaymentCardEntity>> createNewCard({
```

Comment:
I thought we did align on naming the entity, which should not have Entity at the end ?